### PR TITLE
📝 Markdown editor

### PR DESCRIPTION
**Pull request by thesan**
Honestly I didn't expect writing so much code for this feature. When I picked [Slate](https://github.com/ianstormtaylor/slate) I had misunderstood #3403, I thought it was about adding a rich text editor which would save and load markdown just like in Pioneer. Once I realized it was something similar to the GH editor I checked a few alternatives but they weren't very customizable so I kept Slate. But I might have been biased since I really didn't want to restart from scratch several days in.

Otoh it should make adding more feature later on easier if needed.

https://atlas-storybook-git-fork-thesan-feature-markdo-1572b7-joystream.vercel.app/?path=/story/inputs-markdowneditor--markdown-editor

### CODE REVIEW

File: packages/atlas/src/components/_inputs/MarkdownEditor/MarkdownEditor.styles.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+export const ToolBar = styled.div`
+  overflow-x: auto;
+  background-color: ${cVar('colorBackgroundMutedAlpha')};
+  display: flex;
+  padding: ${sizes(2)};
+  gap: ${sizes(2)};
+`
+
+export const StyledFormatButton = styled.button`
```
Comment (WRadoslaw) [inline packages/atlas/src/components/_inputs/MarkdownEditor/MarkdownEditor.styles.ts]: Tertiary variant didn't match the styling?

File: packages/atlas/src/components/_inputs/MarkdownEditor/MarkdownEditor.styles.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  align-items: center;
+  background: none;
+  border: none;
+  cursor: pointer;
+  color: ${cVar('colorCoreNeutral50')};
+  height: ${sizes(8)};
+  width: ${sizes(8)};
+`
+
+export const CharacterCount = styled.div`
```
Comment (WRadoslaw) [inline packages/atlas/src/components/_inputs/MarkdownEditor/MarkdownEditor.styles.ts]: Let's make it `styled(Text)` and handle size by variant prop for consistency

File: packages/atlas/src/components/_inputs/MarkdownEditor/MarkdownEditor.utils.ts [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+
+//
+// Inline format
+//
+
+type InlineFormat = 'emphasis' | 'strong' | 'delete' | 'inlineCode' | 'link'
+function toggleInlineFormat(format: InlineFormat) {
+  return (editor: Editor) => {
+    const initialSelection = editor.selection
+    const expandedSelection = initialSelection && Range.isExpanded(initialSelection) && initialSelection
```
Comment (attemka) [inline packages/atlas/src/components/_inputs/MarkdownEditor/MarkdownEditor.utils.ts:93]: double check for initialSelection

**Review by attemka — APPROVED** [review]
lgtm!

File: packages/atlas/src/components/_inputs/MarkdownEditor/MarkdownEditor.styles.ts (hunk 1 above)
Comment (thesan) [reply packages/atlas/src/components/_inputs/MarkdownEditor/MarkdownEditor.styles.ts]: I just missed it !

**Pull request merged**